### ISSUE TITLE
models/storage: uri is used as directory by storage

### ISSUE THREAD

**Issue by nharraud** [inveniosoftware/invenio-files-rest#43]
**Problem:**
`storage.PyFilesystemStorage.save` [uses the `file.uri` as a directory](https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/storage.py#L173) when it seems that [it is used everywhere else as a file](https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/storage.py#L202).

@lnielsen Is there a use-case where the file's uri is a directory or is this a bug?

**Comment by lnielsen** [comment]
Usage is correct, because ``file.uri`` is passed to ``opendir`` which take the dirname of ``file.uri``.

**Comment by nharraud** [comment]
@lnielsen I just tested it. I set the file's `uri` before saving.
- If the file does not exist, [this line](https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/storage.py#L173) creates a directory matching the full uri, not just the parent directory.
- If the file exists, an error is raised: `fs.errors.ResourceInvalidError: Cannot create directory, there's already a file of that name: b'myfile'`
